From b943b0cb0a0c8360177f07ff5b35075ad73fa28d Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 13 May 2014 16:00:36 +0000
Subject: [PATCH 0540/1077] MANTIS 0032175 extension; Conveyor: conveyor3dev
 send image of baggage to application conveyordev.idl V1.2

- added moduleType IMAGE_SCANNER
- added indicationType IND_PIECE_IMAGE

git-svn-id: svn://localhost/SelfServiceCommon/trunk@672 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/ConveyorDev.idl              | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index d0bd433d..e82411d7 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -39,14 +39,14 @@
 **/
 module conveyordev
 {
-   const string interface_version = "1.1";
+   const string interface_version = "1.2";
 /*
 using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
   RC indicationListener::indication(in any ind);
     with ind: DeviceComponents::indication
       with ind.type:   conveyordev::indicationType
       with ind.status: conveyordev::status
-      with ind.data:   conveyordev::bagInfo                 for IND_PIECE_WEIGHT, IND_PIECE_BARCODE        
+      with ind.data:   conveyordev::bagInfo                 for IND_PIECE_WEIGHT, IND_PIECE_BARCODE, IND_PIECE_IMAGE        
       with ind.data:   conveyordev::characteristics         for IND_CHARACTERISTICS_INFO 
       with ind.data:   conveyordev::extendedCharacteristic  for IND_EXTENDED_CHARACTERISTICS_INFO 
       with ind.data:   DeviceComponents::nullData           for the others
@@ -72,7 +72,8 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
      SCALE,               // get weight of baggage
      BCR,                 // get barcodes of tags on baggage
      RFID_READER,         // get data from rfid-tags on baggage
-     RFID_WRITER          // set data to rfid-tags on baggage
+     RFID_WRITER,         // set data to rfid-tags on baggage
+     IMAGE_SCANNER        // get image of baggage
    };
 
    struct moduleCharacteristic
@@ -342,7 +343,8 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       IND_VERIFICATION_PROCESS_DONE,         /**< Finished CMD_VERIFICATION_PROCESS */
       IND_PARKING_FORWARD_DONE,              /**< Finished CMD_PARKING_FORWARD */
       IND_PARKING_PROCESS_DONE,              /**< Finished CMD_PARKING_PROCESS */
-      IND_PARKING_RETURN_DONE                /**< Finished CMD_PARKING_RETURN */
+      IND_PARKING_RETURN_DONE,               /**< Finished CMD_PARKING_RETURN */
+      IND_PIECE_IMAGE                        /**< image(s) of the piece. */
    };
 
    /**
@@ -352,9 +354,7 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
     *
    **/
    interface Handler : DeviceComponents::Generic
-   {
-   };
-
+   {};
 };
 
 
-- 
2.41.0.windows.1

